test/integration/targets/vmware_host_hyperthreading/tasks/main.yml

```diff
@@ -0,0 +1,83 @@
+# Test code for the vmware_host_acceptance module.
+# Copyright: (c) 2018, Abhijeet Kasurde <akasurde@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# TODO: vcsim does not support Acceptance Level related to operations
+
+- name: store the vcenter container ip
+  set_fact:
+    vcsim: "{{ lookup('env', 'vcenter_host') }}"
+
+- debug: var=vcsim
+
+- name: Wait for Flask controller to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 5000
+    state: started
+
+- name: kill vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/killall
+
+- name: start vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/spawn?cluster=2
+  register: vcsim_instance
+
+- debug:
+    var: vcsim_instance
+
+- name: Wait for vcsim server to come up online
+  wait_for:
+    host: "{{ vcsim }}"
+    port: 443
+    state: started
+
+- name: get a list of hosts from vcsim
+  uri:
+    url: http://{{ vcsim }}:5000/govc_find?filter=H
+  register: hosts
+
+- name: get a host
+  set_fact:
+    host1: "{{ hosts.json[0] | basename }}"
+
+- debug: var=host1
+
+# Hyperthreading optimization is not available for hosts in vcsim
+
+- name: Disable Hyperthreading for a given host
+  vmware_host_hyperthreading:
+    hostname: "{{ vcsim }}"
+    username: "{{ vcsim_instance.json.username }}"
+    password: "{{ vcsim_instance.json.password }}"
+    esxi_hostname: "{{ host1 }}"
+    validate_certs: no
+    enabled: false
+  register: host_hyperthreading_facts
+
+- debug: var=host_hyperthreading_facts
+
+- assert:
+    that:
+      - host_hyperthreading_facts is defined
+      - host_hyperthreading_facts.changed
+
+- name: Disable Hyperthreading for a given host in check mode
+  vmware_host_hyperthreading:
+    hostname: "{{ vcsim }}"
+    username: "{{ vcsim_instance.json.username }}"
+    password: "{{ vcsim_instance.json.password }}"
+    esxi_hostname: "{{ host1 }}"
+    validate_certs: no
+    enabled: false
+  register: host_hyperthreading_facts_check_mode
+  check_mode: yes
+
+- debug: var=host_hyperthreading_facts_check_mode
+
+- assert:
+    that:
+      - host_hyperthreading_facts_check_mode is defined
+      - host_hyperthreading_facts_check_mode.changed
```

Comment:
Add testcases for Cluster with and without check_mode.

Reply:
I cannot enable Hyperthreading on ESXi hosts in vcsim.